### ISSUE TITLE
.note.sgxmeta section is missing in unsigned enclave

### ISSUE THREAD

**Issue by sp21** [google/asylo#54]
I am currently testing non-docker mode of Asylo installation.

I simply ran `bazel run --coinfig=sgx //hello-world` and got the following error
```
ERROR: The enclave image should have '.note.sgxmeta' section Regular sections incorrect The input enclave file is not correct. Error happened while signing the enclave. <EnclaveConfiguration> <ProdID>0</ProdID> <ISVSVN>0</ISVSVN> <StackMaxSize>0x40000</StackMaxSize> <HeapMaxSize>0x47000000</HeapMaxSize> <TCSNum>200</TCSNum> <TCSPolicy>0</TCSPolicy> <DisableDebug>0</DisableDebug> <ProvisionKey>0</ProvisionKey> <EnableKSS>0</EnableKSS> <MiscSelect>0</MiscSelect> <MiscMask>0xffffffff</MiscMask> </EnclaveConfiguration>
```

Actually the `.note.sgxmeta` section header is missing in `libhello_enclave_unsigned.so`
```
$ objdump -h bazel-bin/hello-world/libhello_enclave_unsigned.so
libhello_enclave_unsigned.so:     file format elf64-x86-64

Sections:
Idx Name          Size      VMA               LMA               File off  Algn

  0 .interp       0000000f  0000000000000200  0000000000000200  00000200  2**0
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  1 .note.gnu.property 00000040  0000000000000210  0000000000000210  00000210  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  2 .hash         00000054  0000000000000250  0000000000000250  00000250  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  3 .dynsym       00000180  00000000000002a8  00000000000002a8  000002a8  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  4 .dynstr       000000ea  0000000000000428  0000000000000428  00000428  2**0
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  5 .gnu.version  00000020  0000000000000512  0000000000000512  00000512  2**1
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  6 .gnu.version_d 00000038  0000000000000538  0000000000000538  00000538  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  7 .rela.dyn     00034728  0000000000000570  0000000000000570  00000570  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  8 .rela.plt     000000c0  0000000000034c98  0000000000034c98  00034c98  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  9 .plt          00000090  0000000000034d60  0000000000034d60  00034d60  2**4
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
 10 .text         004569d9  0000000000034e00  0000000000034e00  00034e00  2**6
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
 11 .nipx         00000b36  000000000048b7d9  000000000048b7d9  0048b7d9  2**0
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
 12 malloc_hook   00000212  000000000048c310  000000000048c310  0048c310  2**1
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
 13 .rodata       0008d6c0  000000000048d000  000000000048d000  0048d000  2**12
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
 14 .eh_frame     00118144  000000000051a6c0  000000000051a6c0  0051a6c0  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
 15 .gcc_except_table 0001c001  0000000000632804  0000000000632804  00632804  2**2
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
 16 .tdata        00000010  000000000084e808  000000000084e808  0064e808  2**3
                  CONTENTS, ALLOC, LOAD, DATA, THREAD_LOCAL
 17 .tbss         00000250  000000000084e818  000000000084e818  0064e818  2**3
                  ALLOC, THREAD_LOCAL
 18 .init_array   000004c8  000000000084e818  000000000084e818  0064e818  2**3
                  CONTENTS, ALLOC, LOAD, DATA
 19 .fini_array   00000018  000000000084ece0  000000000084ece0  0064ece0  2**3
                  CONTENTS, ALLOC, LOAD, DATA
 20 .data.rel.ro  00015538  000000000084ed00  000000000084ed00  0064ed00  2**5
                  CONTENTS, ALLOC, LOAD, DATA
 21 .dynamic      000001b0  0000000000864238  0000000000864238  00664238  2**3
                  CONTENTS, ALLOC, LOAD, DATA
 22 .got          00000050  00000000008643e8  00000000008643e8  006643e8  2**3
                  CONTENTS, ALLOC, LOAD, DATA
 23 .got.plt      00000058  0000000000864438  0000000000864438  00664438  2**3
                  CONTENTS, ALLOC, LOAD, DATA
 24 .data         000037ac  00000000008644a0  00000000008644a0  006644a0  2**5
                  CONTENTS, ALLOC, LOAD, DATA
 25 .niprod       000008c0  0000000000867c80  0000000000867c80  00667c80  2**6
                  CONTENTS, ALLOC, LOAD, DATA
 26 .nipd         00000008  0000000000868540  0000000000868540  00668540  2**2
                  CONTENTS, ALLOC, LOAD, DATA
 27 .reserved_data 00100000  0000000000868560  0000000000868560  00668560  2**5
                  CONTENTS, ALLOC, LOAD, DATA
 28 .reserved_bss 00100000  0000000000968560  0000000000968560  00768560  2**5
                  CONTENTS, ALLOC, LOAD, DATA
 29 .reserved_heap 00100000  0000000000a68560  0000000000a68560  00868560  2**5
                  CONTENTS, ALLOC, LOAD, DATA
 30 .bss          0001a468  0000000000b68580  0000000000b68580  00968560  2**6
                  ALLOC
 31 .comment      00000011  0000000000000000  0000000000000000  00968560  2**0
                  CONTENTS, READONLY
```

To my understanding, the `.note.sgxmeta` section is derived from `libsgx_trts.a` library.

AS the header is not found, there may be some problem with linking TRTS library?

Since I am not a bazel expert, it is hard for me to pinpoint the root cause

**Comment by KeithMoyer** [comment]
Thanks, @shpark3.

We also found the same issue recently, have root caused it, fixed it internally, and will be pushing out a fix for it soon.

This is related to how Bazel 1.0+ links things. In the meantime, you can either:
a) use the gcr.io/asylo-framework/asylo:buildenv-v0.4.1 docker image (this had an older version of Bazel)
b) add alwayslink = 1 to the trts_hw target in the [SGX SDK patch](https://github.com/google/asylo/blob/v0.5.0/asylo/distrib/sgx_x86_64/linux_sgx_2_6.patch#L3888)

**Comment by KeithMoyer** [comment]
This was fixed by f5651392cf822e255e7de93956503c0179d4c021
